services/app-web/src/hooks/useTealium.ts

```diff
@@ -0,0 +1,104 @@
+import { useEffect } from 'react'
+import { usePage } from '../contexts/PageContext'
+import { useCurrentRoute } from './useCurrentRoute'
+import { createScript } from './useScript'
+import { PageTitlesRecord } from '../constants/routes'
+import { useAuth } from '../contexts/AuthContext'
+import { getTealiumEnv, CONTENT_TYPE_BY_ROUTE } from '../constants/tealium'
+
+/*
+Tealium is the data layer for Google Analytics and other data tracking at CMS
+    The hooks in this file have two purposes:
+    1. Loads JavaScript code called utag.js into the application. This contains all of the generated code necessary to use Tealium and third-party tags.
+    2. Tracks new page views using the Tealium Universal Data Format, defined in this doc: https://docs.tealium.com/platforms/javascript/universal-data-object/
+*/
+const useTealium = (): void => {
+    const { currentRoute, pathname } = useCurrentRoute()
+    const { heading } = usePage()
+    const { loggedInUser } = useAuth()
+
+    useEffect(() => {
+        // Do not add tealium for local dev
+        if (process.env.REACT_APP_STAGE_NAME === 'local') return
+
+        const tealiumEnv = getTealiumEnv(
+            process.env.REACT_APP_STAGE_NAME || 'main'
+        )
+        const tealiumProfile = 'cms-mcreview'
+        if (!tealiumEnv || !tealiumProfile) {
+            console.error(
+                `Missing key configuration for Tealium. tealiumEnv: ${tealiumEnv} tealiumProfile: ${tealiumProfile}`
+            )
+        }
+
+        // Suppress automatic page views for SPA
+        window.utag_cfg_ovrd = window.utag_cfg_ovrd || {}
+        window.utag_cfg_ovrd.noview = true
+
+        // Load utag.sync.js - add to head element - SYNC load from src
+        const initializeTagManagerSnippet = createScript({
+            src: `https://tags.tiqcdn.com/utag/cmsgov/${tealiumProfile}/${tealiumEnv}/utag.sync.js`,
+            id: 'tealium-load-tags-sync',
+        })
+        document.head.appendChild(initializeTagManagerSnippet)
+
+        // Load utag.js - Add to body element- ASYNC load inline script
+        const inlineScript =
+            document.createTextNode(`(function (t, e, a, l, i, u, m) {
```

Comment:
wut

Reply:
lol I know. If you look at both the docs from Tealium and the CMS page level tracking instructions - they both say load the script this way (as an inline anonymouse script) so that's why I did it.  The `t,e,a,l,i,u,m` stuff is from the CMS setup specifically.